modules/battery_manager/php/testprovisioner.class.inc

```diff
@@ -0,0 +1,75 @@
+<?php
+/**
+ * This file implements a data provisioner to get all possible tests
+ * for the battery_manager menu page.
+ *
+ * PHP Version 7
+ *
+ * @category   Behavioural
+ * @package    Main
+ * @subpackage Imaging
+ * @author     Henri Rabalais <henri.rabalais@mcin.ca>
+ * @license    http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link       https://www.github.com/aces/Loris/
+ */
+
+namespace LORIS\battery_manager;
+
+/**
+ * This class implements a data provisioner to get all possible tests
+ * for the battery_manager menu page.
+ *
+ * PHP Version 7
+ *
+ * @category   Behavioural
+ * @package    Main
+ * @subpackage Imaging
+ * @author     Henri Rabalais <henri.rabalais@mcin.ca>
+ * @license    http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link       https://www.github.com/aces/Loris/
+ */
+class TestProvisioner extends \LORIS\Data\Provisioners\DBRowProvisioner
+{
+    /**
+     * Select all tests from the test_battery table that match the attribute
+     * values passed as conditions and will return an array of test objects
+     * indexed by id.
+     *
+     * @return array $tests List of Test Objects that match the query
+     */
+    public function __construct()
+    {
+        parent::__construct(
+            "SELECT b.id,
+               t.Test_name as testName,
+               b.AgeMinDays as ageMinDays,
+               b.AgeMaxDays as ageMaxDays,
+               b.Stage as stage,
+               s.SubprojectID as subproject,
+               b.Visit_label as visitLabel,
+               p.CenterID as centerId,
+               b.firstVisit,
+               b.instr_order as instrumentOrder,
+               b.Active as active
+        FROM test_battery b
+          LEFT JOIN test_names t USING (Test_name)
+          LEFT JOIN subproject s USING (SubprojectID)
+          LEFT JOIN psc p USING (CenterID)
+        ORDER BY t.Full_name",
```

Comment:
I'm not sure why, but this doesn't seem to be working. Every time the query is made, the rows appear in a different order.

Reply:
@HenriRabalais Have you tried sorting by a column that is part of the result rows? ex: testName

Reply:
I'll give it a try